banyand/query/processor.go

```diff
@@ -160,6 +161,62 @@ func (p *measureQueryProcessor) Rev(ctx context.Context, message bus.Message) (r
 		resp = bus.NewMessage(bus.MessageID(now), common.NewError("invalid event data type"))
 		return
 	}
+	resp = p.executeQuery(ctx, queryCriteria)
+
+	if queryCriteria.RewriteAggTopNResult {
+		result := resp.Data().(*measurev1.QueryResponse).DataPoints
+		rewriteResult := make([]*measurev1.DataPoint, 0, len(result))
+		entities := make([]map[string]*modelv1.TagValue, 0, len(result))
+		for _, dp := range result {
+			entity := make(map[string]*modelv1.TagValue)
+			for _, tagFamily := range dp.GetTagFamilies() {
+				for _, tag := range tagFamily.GetTags() {
+					entity[tag.GetKey()] = tag.GetValue()
+				}
+			}
+			if len(entity) > 0 {
+				entities = append(entities, entity)
+			}
+		}
+		rewriteQueryCriteria := &measurev1.QueryRequest{
+			Groups:          queryCriteria.Groups,
+			Name:            queryCriteria.Name,
+			TimeRange:       queryCriteria.TimeRange,
+			Criteria:        queryCriteria.Criteria,
```

Comment:
You should update `Criteria` like 

```yaml
criteria:
  le:
    op: "LOGICAL_OP_AND"
    left:
      condition:
        name: "tag1_in_groupBy"
        op: "BINARY_OP_IN"
        value:
          strArray:
            value: ["a", "b", "c"]
     right:
      condition:
        name: "tag2_in_groupBy"
        op: "BINARY_OP_IN"
        value:
          strArray:
            value: ["1", "2", "3"]
```

If there is only a tag in groupBy fileld, the expression can be simplied:

```yaml
criteria:
  condition:
    name: "tag1_in_groupBy"
    op: "BINARY_OP_IN"
    value:
      strArray:
        value: ["a", "b", "c"]
```